framework/core/src/Discussion/DiscussionRepository.php

```diff
@@ -28,40 +28,39 @@ public function query()
      * Find a discussion by ID, optionally making sure it is visible to a
      * certain user, or throw an exception.
      *
-     * @param int $id
-     * @param User $user
+     * @param int|string $id
+     * @param User|null $user
      * @return \Flarum\Discussion\Discussion
      */
     public function findOrFail($id, User $user = null)
     {
-        $query = Discussion::where('id', $id);
+        $query = $this->query()->where('id', $id);
 
         return $this->scopeVisibleTo($query, $user)->firstOrFail();
     }
 
     /**
      * Get the IDs of discussions which a user has read completely.
      *
-     * @deprecated 1.3 Use `getReadIdsQuery` instead
-     *
      * @param User $user
-     * @return array
+     * @return \Illuminate\Database\Eloquent\Collection<Discussion>
```

Comment:
Should this be imported instead of using the fully qualified name?

Reply:
phpDocs seem to be the exception looking at our codebase